include/NovelRT/Input/Glfw/GlfwInputDevice.h

```diff
@@ -0,0 +1,42 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+
+#ifndef NOVELRT_INPUT_GLFW_GLFWINPUTDEVICE_H
+#define NOVELRT_INPUT_GLFW_GLFWINPUTDEVICE_H
+
+#ifndef NOVELRT_INPUT_GLFW_H
+#error NovelRT does not support including types explicitly by default. Please include Input.Glfw.h instead for the Input::Glfw namespace subset.
+#endif
+
+namespace NovelRT::Input::Glfw
+{
+    class GlfwInputDevice final : public Input::IInputDevice
+    {
+    private:
+        bool _isInitialised;
+        GLFWwindow* _window;
```

Comment:
Is there a reason we're not just passing in the smart pointer the windowing device provides?

Reply:
Do you mean a smart pointer to the Windowing Device itself?
At the time, it was just easier to allow taking in a `void*` via the `Initialise` call and yelling if it's not actually a `GLFWwindow*`, so that way we can consistently reuse the existing `GLFWwindow*` in the update loop

Reply:
Discord - this is a "will fix later"